internal/federation/handle.go

```diff
@@ -277,3 +279,101 @@ func HandleMediaRequests(mediaIds map[string]func(w http.ResponseWriter)) func(*
 		mediamux.Handle("/r0/download/{origin}/{mediaId}", downloadFn).Methods("GET")
 	}
 }
+
+// HandleTransactionRequests is an option which will process GET /_matrix/federation/v1/send/{transactionID} requests universally when requested.
+func HandleTransactionRequests() func(*Server) {
```

Comment:
Might be cute to have a callback function here which is passed with valid `gmsl.Event` structs, since it's fairly common to want to process stuff on `/send`. E.g:
```go
func HandleTransactionRequests(cb func(gmsl.Event)) func(*Server) {
    // when ready..
    cb(event)
}
```

Reply:
This is what I was thinking, yeah. Then we could perhaps augment `TestOutboundFederationSend` to use this instead, while slimming down the test at the same time.

Reply:
Shall I add another callback method for EDUs?